roles/translator/src/upstream_sv2/upstream.rs

```diff
@@ -215,21 +219,39 @@ impl Upstream {
     /// Parses the incoming SV2 message from the Upstream role and routes the message to the
     /// appropriate handler.
     pub fn parse_incoming(self_: Arc<Mutex<Self>>) {
+        let (
+            tx_frame,
+            tx_sv2_extranonce,
+            tx_sv2_new_ext_mining_job,
+            tx_sv2_set_new_prev_hash,
+            recv,
+            tx_status,
+        ) = self_
+            .safe_lock(|s| {
+                (
+                    s.connection.sender.clone(),
+                    s.tx_sv2_extranonce.clone(),
+                    s.tx_sv2_new_ext_mining_job.clone(),
+                    s.tx_sv2_set_new_prev_hash.clone(),
+                    s.connection.receiver.clone(),
+                    s.tx_status.clone(),
+                )
+            })
+            .unwrap();
```

Comment:
Are we not doing the map_err() here because it's unlikely that the clones will fail?

Reply:
This section of code runs on the main thread on startup so there is little chance that it will panic and if it does the main thread will panic, but I can convert it to return a result and we can handle it on the main thread